### PR TITLE
Added a simple way to get a player by their xuid

### PR DESCRIPTION
**Pull request by qlow**
Simple method to get a player by their Xbox UUID. Can be used by Bungee plugins that need to access the GeyserSession

### CODE REVIEW

**Comment by Tim203** [comment]
While this method won't be called a lot, I still recommend using plain for loops

**Comment by qlow** [comment]
Any reasons for that?

**Comment by Tim203** [comment]
For loops are better for performance.
While it might look fancy, I prefer performance xd

**Comment by rtm516** [comment]
Might also be better to name it `GeyserConnector#getSessionByXboxUuid` since that's more descriptive.
Also please add a JavaDoc comment to the method and if you can find the few for loops in the code that do this and replace them with this method. I think there are ones in `SkinUtils.java` and some of the commands

**Comment by qlow** [comment]
> Might also be better to name it `GeyserConnector#getSessionByXboxUuid` since that's more descriptive.
> Also please add a JavaDoc comment to the method and if you can find the few for loops in the code that do this and replace them with this method. I think there are ones in `SkinUtils.java` and some of the commands

I named this method getPlayer... because the corresponding methods are named "addPlayer" and "removePlayer". But I can easily change the name of it and replace the for loops.

**Comment by rtm516** [comment]
> I named this method getPlayer... because the corresponding methods are named "addPlayer" and "removePlayer".

Ah your correct, leave the method name as is for now then.

**Comment by qlow** [comment]
> > I named this method getPlayer... because the corresponding methods are named "addPlayer" and "removePlayer".
> 
> Ah your correct, leave the method name as-is for now then.

And just found out that nowhere the player is retrieved by the XUID, only by the UUID of the player entity. Should I create an extra method for that?

**Comment by rtm516** [comment]
> And just found out that nowhere the player is retrieved by the XUID, only by the UUID of the player entity. Should I create an extra method for that?

If you don't mind, would make sense to roll it into this PR since they are very simmilar.

**Comment by qlow** [comment]
done

File: connector/src/main/java/org/geysermc/connector/GeyserConnector.java [hunk 1]
```diff
@@ -325,6 +322,40 @@ public void removePlayer(GeyserSession player) {
...
+     *
+     * @param uuid the uuid
+     * @return the player or <code>null</code> if there is no player online with this UUID
+     */
+    public GeyserSession getPlayerByUuid(UUID uuid) {
+        for (GeyserSession session : players) {
+            if (!session.getPlayerEntity().getUuid().equals(uuid))
+                continue;
+
+            return session;
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/GeyserConnector.java:336]:
```suggestion
            if (session.getPlayerEntity().getUuid().equals(uuid)) {
                return session;
            }
```

File: connector/src/main/java/org/geysermc/connector/GeyserConnector.java [hunk 2]
```diff
@@ -325,6 +322,40 @@ public void removePlayer(GeyserSession player) {
...
+     *
+     * @param xboxUuid the Xbox UUID
+     * @return the player or <code>null</code> if there is no player online with this XUID
+     */
+    public GeyserSession getPlayerByXboxUuid(String xboxUuid) {
+        for (GeyserSession session : players) {
+            if (session.getAuthData() == null || !session.getAuthData().getXboxUUID().equals(xboxUuid))
+                continue;
+
+            return session;
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/GeyserConnector.java:353]:
```suggestion
            if (session.getAuthData() != null && session.getAuthData().getXboxUUID().equals(xboxUuid))
                return session;
            }
```

**Review by rtm516 — CHANGES_REQUESTED** [review]

**Comment by rtm516** [comment]
Can you also add `{}` to some of the other if statements to follow the styling of the project?

File: connector/src/main/java/org/geysermc/connector/GeyserConnector.java [hunk 3]
```diff
@@ -325,6 +322,38 @@ public void removePlayer(GeyserSession player) {
...
+        return null;
+    }
+
+    /**
+     * Gets a player by their Xbox UUID
+     *
+     * @param xboxUuid the Xbox UUID
+     * @return the player or <code>null</code> if there is no player online with this XUID
+     */
+    public GeyserSession getPlayerByXboxUuid(String xboxUuid) {
```
Comment (lukeeey) [inline connector/src/main/java/org/geysermc/connector/GeyserConnector.java:347]: Name it `getPlayerByXuid`

**Review by lukeeey — DISMISSED** [review]

File: connector/src/main/java/org/geysermc/connector/GeyserConnector.java (hunk 3 above)
Comment (lukeeey) [reply connector/src/main/java/org/geysermc/connector/GeyserConnector.java:347]: **U**niversally **U**nique **ID**dentifier
**X**box **U**ser **ID**entifier (or something)

Comment (qlow) [reply connector/src/main/java/org/geysermc/connector/GeyserConnector.java:347]: I named it xboxuuid because it's named like this in AuthData

Comment (lukeeey) [reply connector/src/main/java/org/geysermc/connector/GeyserConnector.java:347]: It shouldn't be

Comment (Camotoy) [reply connector/src/main/java/org/geysermc/connector/GeyserConnector.java:347]: If you could make this change, then this PR is good to go.

**Review by Camotoy — APPROVED** [review]

**Review by Camotoy — APPROVED** [review]

**Pull request merged**